TeamCode/src/main/java/org/firstinspires/ftc/teamcode/api/Robot.java

```diff
@@ -2,73 +2,157 @@
+    public void adjustClaw(double amount) {
+        // Version 1
+        this.clawLeft.setPosition(this.clawLeft.getPosition() + amount);
+        this.clawRight.setPosition(this.clawRight.getPosition() + amount);
+
+        // Version 2
+
+        /*
+        * double finalPosition = this.clawLeft.getPosition() + amount;
+        * this.clawLeft.setPosition(finalPosition);
+        * this.clawRight.setPosition(finalPosition);
+        */
     }
 
     @Override
-    public void moveFront(double power) {
-        this.frontLeft.setPower(power);
-        this.frontRight.setPower(power);
+    public void openClaw() {
+        this.positionClaw(0.9);
     }
 
     @Override
-    public void moveBack(double power) {
-        this.backLeft.setPower(power);
-        this.backRight.setPower(power);
+    public void closeClaw() {
+        this.positionClaw(0.2);
     }
 
-    public void spinDuck(double power) {
+    @Override
+    public void powerDuck(double power) {
         this.duck.setPower(power);
     }
 
-    public void setMotorsDirection(@NonNull DcMotor[] motors, DcMotor.Direction direction) {
-        for (DcMotor motor: motors) {
-            motor.setDirection(direction);
+    @Override
+    public void initTeleOp(@NonNull OpMode opMode) {
+        // For future reference
+        this.opMode = opMode;
+
+        // InitRobot has an executor service, not sure if that should be included here
+
+        // Motors
+        this.bl = new SmartMotor(opMode.hardwareMap.get(DcMotorEx.class, Naming.MOTOR_BL));
+        this.br = new SmartMotor(opMode.hardwareMap.get(DcMotorEx.class, Naming.MOTOR_BR));
+        this.fl = new SmartMotor(opMode.hardwareMap.get(DcMotorEx.class, Naming.MOTOR_FL));
+        this.fr = new SmartMotor(opMode.hardwareMap.get(DcMotorEx.class, Naming.MOTOR_FR));
+
+        this.duck = new SmartMotor(opMode.hardwareMap.get(DcMotorEx.class, Naming.MOTOR_DUCK));
+
+        // Servos
+        this.armLeft = new SmartServo(opMode.hardwareMap.get(Servo.class, Naming.SERVO_ARM_LEFT));
+        this.armRight = new SmartServo(opMode.hardwareMap.get(Servo.class, Naming.SERVO_ARM_RIGHT));
+        this.clawLeft = new SmartServo(opMode.hardwareMap.get(Servo.class, Naming.SERVO_CLAW_LEFT));
+        this.clawRight = new SmartServo(opMode.hardwareMap.get(Servo.class, Naming.SERVO_CLAW_RIGHT));
+
+        /*
+        // Sensors
+        this.parkSensor = new SmartColorSensor(
+                (NormalizedColorSensor) opMode.hardwareMap.get(ColorSensor.class, Naming.COLOR_SENSOR_PARK)
+        );
+        this.webcam0 = opMode.hardwareMap.get(WebcamName.class, Naming.WEBCAM_0);
+        this.gyro0 = new Gyroscope(opMode.hardwareMap.get(BNO055IMU.class, Naming.GYRO_0), Naming.GYRO_0);
+        this.gyro1 = new Gyroscope(opMode.hardwareMap.get(BNO055IMU.class, Naming.GYRO_1), Naming.GYRO_1);
+        */
+
+        // Looped Config
+        SmartMotor[] wheelLoop = new SmartMotor[]{this.fl, this.fr, this.bl, this.br};
```

Comment:
Perhaps use a hashmap for the motors instead so that we don't have to keep track of indexes and instead can use strings?

Reply:
Hash maps are a pain to work with in Java. If it were Python, the I’d probably say yes. For Java, though, it seems better to use direct variables and avoid Nonnulls. Worst case scenario, we’ll just refactor in Android Studio.